Zinc/Private/ZincContentBundleDeleteTask.m

```diff
@@ -0,0 +1,294 @@
+    [inputStream open];
+    NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithStream:inputStream
+                                                               options:0
+                                                                 error:&error];
+    [inputStream close];
+
+    if (error) {
+        NSLog(@"Error while converting json file to NSDictionary: %@", repoJSONPath);
+        return nil;
+    }
+
+    return jsonDict;
+}
+
+/* return YES if success, NO otherwise
+ */
+- (BOOL)writeToRepoJSONWithJSONDict:(NSDictionary *)jsonDict {
+    [self deleteFileAtURL:self.repoJSONURL];
+
+    NSString *repoJSONPath = [self repoJSONPath];
+    NSError *error;
+
+    NSOutputStream *outputStream = [[NSOutputStream alloc] initToFileAtPath:repoJSONPath
+                                                                     append:NO];
+    [outputStream open];
+    [NSJSONSerialization writeJSONObject:jsonDict
+                                toStream:outputStream
+                                 options:0
+                                   error:&error];
+    [outputStream close];
+
+    if (error) {
+        NSLog(@"couldn't write to file: %@", repoJSONPath);
+    }
+
+    return !error;
+}
+
+- (BOOL)deleteFromRepoJSONBundleEntriesWithBundleIDs:(NSSet<NSString *> *)bundleIDs {
+    NSMutableDictionary *repoJSONDict = [[self repoJSONDict] mutableCopy];
+    if (repoJSONDict == nil) {
+        return NO;
+    }
+
+    NSString * const bundlesKey = @"bundles";
+    NSDictionary<NSString *, id> *originalBundlesByID = repoJSONDict[bundlesKey];
+    NSMutableDictionary<NSString *, id> *newBundlesByID = [NSMutableDictionary new];
+
+    [originalBundlesByID enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull bundleID,
+                                                             id _Nonnull value,
+                                                             BOOL * _Nonnull stop) {
+        if (![bundleIDs containsObject:bundleID]) {
+            newBundlesByID[bundleID] = value;
+        }
+    }];
+
+    repoJSONDict[bundlesKey] = [newBundlesByID copy];
+
+    return [self writeToRepoJSONWithJSONDict:repoJSONDict];
+}
+
+#pragma mark - URLs
+
+- (NSURL *)urlForSearchPathDirectory:(NSSearchPathDirectory)searchPathDirectory {
+    NSURL *url = [NSFileManager.defaultManager URLsForDirectory:searchPathDirectory inDomains:NSUserDomainMask][0];
+    NSAssert(url != nil, @"url not found for NSSearchPathDirectory: %ld", (long)searchPathDirectory);
+    return url;
+}
+
+- (NSURL *)absoluteURLForPathRelativeToZincFolder:(NSString *)path isDirectory:(BOOL)isDirectory {
+    NSURL *docDirURL = [self urlForSearchPathDirectory:NSDocumentDirectory];
+    NSURL *zincDirURL = [docDirURL URLByAppendingPathComponent:@"zinc"];
+    NSURL *url = [zincDirURL URLByAppendingPathComponent:path
+                                             isDirectory:isDirectory];
+    NSParameterAssert(url);
+    return url;
+
+}
+
+- (NSURL *)bundlesFolderURL {
```

Comment:
any chance you can use ZincRepo's bundlesPath for this? same question with the others